packages/app/tasks/get-pool-pending-rewards.js

```diff
@@ -0,0 +1,318 @@
+                        const newReserveAmount = returnValues._newReserveAmount;
+                        const prevPoolAmount = returnValues._prevPoolAmount;
+                        const newPoolAmount = returnValues._newPoolAmount;
+
+                        trace(
+                            'Found ProtectionUpdated event at block',
+                            arg('blockNumber', blockNumber),
+                            arg('provider', provider),
+                            arg('prevPoolAmount', prevPoolAmount),
+                            arg('newPoolAmount', newPoolAmount),
+                            arg('prevReserveAmount', prevReserveAmount),
+                            arg('newReserveAmount', newReserveAmount),
+                            arg('timestamp', timestamp),
+                            arg('tx', transactionHash)
+                        );
+
+                        // Try to find the pool and reserves tokens by matching the position in a previous block.
+                        // Please note that we are assuming that a single position wasn't added and removed in the
+                        // same block.
+                        const matches = [];
+                        const prevBlock = blockNumber - 1;
+                        let ids = await web3Provider.call(
+                            contracts.LiquidityProtectionStore.methods.protectedLiquidityIds(provider),
+                            {},
+                            prevBlock
+                        );
+                        for (const id of ids) {
+                            const position = await getPosition(id, prevBlock);
+                            if (
+                                new BN(position.reserveAmount).eq(new BN(prevReserveAmount)) &&
+                                new BN(position.poolAmount).eq(new BN(prevPoolAmount))
+                            ) {
+                                matches.push({
+                                    poolToken: position.poolToken,
+                                    reserveToken: position.reserveToken
+                                });
+                            }
+                        }
+
+                        if (matches.length === 0) {
+                            warning(
+                                'Failed to fully match pool and reserve tokens. Trying to look for an updated position in the same block (assuming no more than a two updates in the same block)'
+                            );
+
+                            ids = await web3Provider.call(
+                                contracts.LiquidityProtectionStore.methods.protectedLiquidityIds(provider),
+                                {},
+                                blockNumber
+                            );
+                            for (const id of ids) {
+                                const position = await getPosition(id, blockNumber);
```

Comment:
Consider `Promise.all` outside the `for` loop...

Reply:
Similar to the above, I don't want to over-optimize these scripts on purpose, since they are running against a local node and produce crucial data.